worlds/tunic/__init__.py

```diff
@@ -74,6 +95,56 @@ def generate_early(self) -> None:
                 self.options.entrance_rando.value = passthrough["entrance_rando"]
                 self.options.shuffle_ladders.value = passthrough["shuffle_ladders"]
 
+    @classmethod
+    def stage_generate_early(cls, multiworld: MultiWorld) -> None:
+        tunic_worlds: Tuple[TunicWorld] = multiworld.get_game_worlds("TUNIC")
+        for tunic in tunic_worlds:
+            if isinstance(tunic.options.entrance_rando.value, str):
+                group = tunic.options.entrance_rando.value
+                # if this is the first world in the group, set the rules equal to its rules
+                if group not in cls.seed_groups:
+                    cls.seed_groups[group] = SeedGroup(logic_rules=tunic.options.logic_rules.value,
+                                                       laurels_at_10_fairies=tunic.options.laurels_location == 3,
+                                                       fixed_shop=bool(tunic.options.fixed_shop),
+                                                       plando=multiworld.plando_connections[tunic.player])
+                else:
+                    # lower value is more restrictive
+                    if tunic.options.logic_rules.value < cls.seed_groups[group]["logic_rules"]:
+                        cls.seed_groups[group]["logic_rules"] = tunic.options.logic_rules.value
+                    # laurels at 10 fairies changes logic for secret gathering place placement
+                    if tunic.options.laurels_location == 3:
+                        cls.seed_groups[group]["laurels_at_10_fairies"] = True
+                    # fewer shops, one at windmill
+                    if tunic.options.fixed_shop:
+                        cls.seed_groups[group]["fixed_shop"] = True
+
+                    if multiworld.plando_connections[tunic.player]:
+                        # loop through the connections in the player's yaml
+                        for cxn in multiworld.plando_connections[tunic.player]:
+                            new_conn = True
+                            # check if either of the entrances in the pair match, then check for conflicts
+                            for group_cxn in cls.seed_groups[group]["plando"]:
+                                if cxn.entrance == group_cxn.entrance:
```

Comment:
This seems a bit more complex than necessary, the Exception text is the same both times so this could just be an `if thing or other_thing`, right?

Reply:
```
                            for group_cxn in cls.seed_groups[group]["plando"]:
                                if not (cxn.entrance == group_cxn.entrance or cxn.entrance == group_cxn.exit):
                                    continue
                                
                                new_conn = False
                                
                                is_mismatched = (
                                        cxn.entrance == group_cxn.entrance and cxn.exit != group_cxn.exit
                                        or cxn.entrance == group_cxn.exit and cxn.exit != group_cxn.entrance
                                )
                                
                                if is_mismatched:
                                    raise Exception(f"TUNIC: Conflict between seed group {group}'s plando "
                                                    f"connection {group_cxn.entrance} <-> {group_cxn.exit} and "
                                                    f"{tunic.multiworld.get_player_name(tunic.player)}'s plando "
                                                    f"connection {cxn.entrance} <-> {cxn.exit}")
```

How we feeling about this? We are doing a redundant check now but it looks a lot better and has a lot fewer conditionals